recipe/meta.yaml

```diff
@@ -14,26 +13,22 @@ source:
     - GEOSSTRtreeTest.cpp.patch  # [win]
 
 build:
-  number: 0
+  number: 1
   features:
-    - vc9  # [win and py27]
-    - vc10  # [win and py34]
-    - vc14  # [win and py>=35]
+    - vc{{ vc }}  # [win]
+  run_exports:
+    # pretty poor backcompat.  SO name changes each release.
+    #   https://abi-laboratory.pro/tracker/timeline/geos/
+    - {{ pin_subpackage('geos', max_pin='x.x.x') }}
 
 requirements:
   build:
-    - python  # [win]
-    - msinttypes  # [win]
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
```

Comment:
Linux seems happy with this change and everything worked with the image [without](https://github.com/conda-forge/geos-feedstock/pull/25/commits/5877962abc1bd6f817dc09310cc43019f5c50807) a pre-installed compiler.

```shell
geos-3.6.0-hd75213b_1:
    libgeos-3.6.0.so (lib/libgeos-3.6.0.so)

libstdcxx_linux-64-7.1.0-hddd74c5_0:
    libstdc++.so.6 (lib/libstdc++.so.6)

system:
    libc.so.6 (/lib64/libc.so.6)
    libgcc_s.so.1 (/lib64/libgcc_s.so.1)
    libm.so.6 (/lib64/libm.so.6)
    linux-vdso.so.1 ()
```

`OS X` is  not happy when I defined `{{ compiler('cxx') }}`. I guess I need to make additional changes to [`conda_build_config.yaml`](https://github.com/AnacondaRecipes/aggregate/blob/master/conda_build_config.yaml)...

Reply:
OSX kind of sucks.  You need to get the 10.9 SDK from somewhere (perhaps https://github.com/phracker/MacOSX-SDKs).  Extract it into /opt, so that you have the /opt/MacOSX10.9.sdk.  Then you should be good with our conda_build_config.yaml, unmodified.

Alternatively, change conda_build_config.yaml to point at the root of that SDK if it exists somehwere else on the system.

Reply:
> Alternatively, change conda_build_config.yaml to point at the root of that SDK if it exists somewhere else on the system.

I was thinking about that solution, so can use what is already in Travis-CI. Not sure if that is the best approach though. My Mac knowledge is zero.

Reply:
My understanding is that downloading the SDK we would have more control and we could "ignore" Travis-CI updates more easily, but that would increase the build time...

Reply:
If you know that the system is running with 10.9 you can get away with setting CONDA_BUILD_SYSROOT to `/`. I would avoid not setting it as I have modified the behaviour of the macOS linker if it finds `CONDA_BUILD_SYSROOT` is set (it looks for our own build of libc++ in the right place instead of finding the old system one basically - 10.9's libc++ is pretty old now and doesn't support all the new language features).

Reply:
> If you know that the system is running with 10.9 you can get probably away with setting `CONDA_BUILD_SYSROOT` to `/`. I would avoid not setting it as I have modified the behaviour of the macOS linker if it finds `CONDA_BUILD_SYSROOT` is set (it looks for our own build of libc++ in the right place instead of finding the old system one basically - 10.9's libc++ is pretty old now and doesn't support all the new language features).

@mingwandroid I won't pretend I get all the consequences of the options we have here. I'd definitely would like to go the path where we have more control, and it seems that ignoring the system's stuff and setting it to find the newer version is better.

We can talk more about that in a meeting some day. I guess we cannot advance this anyways until `c3i` packages are in `defaults`.

Reply:
You can use `c3i_test` to get an idea of where things stand. You need to be using very recent `condas` and git master of `conda-build` at present though.

Happy to go through the design for the compilers. We should arrange a meeting sometime? Probably good to get a few people involved too as I see that `conda-forge` is forging (ha ha) its own path and it'd be really good if we can work towards binary compatibility as well as shared recipe curation.

Reply:
> You can use `c3i_test` to get an idea of where things stand.

I am testing things here with small bite size [commits](https://github.com/conda-forge/geos-feedstock/pull/25/commits). I added `conda-build=3` in https://github.com/conda-forge/geos-feedstock/pull/25/commits/3a7f3fc134b6dec88504652b41cda3067835d9d0 and `ci3_test` right after that in https://github.com/conda-forge/geos-feedstock/pull/25/commits/f19a0d9f7fe2659720a163d6fbe5c385c5ea220c (Linux that worked just fine BTW :smile:)

We did not get `conda 4.3.25` but we got close [enough](https://travis-ci.org/conda-forge/geos-feedstock/builds/269141586#L223):

```shell
               platform : osx-64
          conda version : 4.3.22
       conda is private : False
      conda-env version : 4.3.22
    conda-build version : 3.0.13
         python version : 3.6.1.final.0
       requests version : 2.14.2
       root environment : /Users/travis/miniconda3  (writable)
    default environment : /Users/travis/miniconda3
       envs directories : /Users/travis/miniconda3/envs
                          /Users/travis/.conda/envs
          package cache : /Users/travis/miniconda3/pkgs
                          /Users/travis/.conda/pkgs
           channel URLs : https://conda.anaconda.org/c3i_test/osx-64
                          https://conda.anaconda.org/c3i_test/noarch
                          https://conda.anaconda.org/conda-forge/osx-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.continuum.io/pkgs/free/osx-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/r/osx-64
                          https://repo.continuum.io/pkgs/r/noarch
                          https://repo.continuum.io/pkgs/pro/osx-64
                          https://repo.continuum.io/pkgs/pro/noarch
            config file : /Users/travis/.condarc
             netrc file : None
           offline mode : False
             user-agent : conda/4.3.22 requests/2.14.2 CPython/3.6.1 Darwin/14.5.0 OSX/10.10.5    
                UID:GID : 501:20
```

I'll force install `conda 4.3.25` too in my next commit.

Reply:
> You need to get the 10.9 SDK from somewhere...

This is already included on the Travis CI VM we are using. If it is in XCode, it should be something like `/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk`. Though it may not be on Travis CI (not sure). Running `xcodebuild -sdk -version` should provide info about all of the SDKs available.

Reply:
The question is not where is the location on `Travis-CI` but rather if we are going to use it or not.

Reply:
From my perspective if `/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk` is the same as the other one you can find on github then either will work equivalently (I guess that's obvious).

Using a fixed known SDK is a very good thing, but it should be noted that if you set `CONDA_BUILD_SYSROOT` and are using my fork of the Apple linker and clang then I do special things to ensure that the correct libraries and headers get found.

Reply:
> From my perspective if `/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk` is the same as the other one you can find on github then either will work equivalently (I guess that's obvious).

Indeed. I was thinking more on that time when Travis-CI update the images and we were forced to update as well. I guess that this is a minor point about control of the SDK vs download time. No need to get into this now.

> but it should be noted that if you set `CONDA_BUILD_SYSROOT` and are using my fork of the Apple linker and clang then I do special things to ensure that the correct libraries and headers get found.

I am more interested in experimenting with this :wink:

Reply:
The concern with taking it from GitHub is whether this is permitted by the license. I don't know the answer to that question. IANAL.

> Using a fixed known SDK is a very good thing...

Agreed. It doesn't matter what compiler we end up using. We still need to be doing this.

Would love to talk about what you are doing with `clang` as well, but maybe that should be saved for another discussion.

Reply:
> Agreed. It doesn't matter what compiler we end up using. We still need to be doing this.

Yeah, I know what you are saying, and that you didn't mean it that way, but for binary compatibility it matters greatly what compiler is used. Hopefully we can standardise on mine. They are based on very recent clang 4.0.1 and GCC 7.1.0 and should therefore be good for a many years to come.

Reply:
> ...for binary compatibility it matters greatly what compiler is used

Of course. Was only meaning that fixing the SDK is important regardless, but I think you picked up on this already.